### ISSUE TITLE
Creating a db home with no database

### ISSUE THREAD

**Issue by lucianf** [oracle/terraform-provider-oci#1219]
It seems inconsistent that with the OCI console and OCI CLI (hence presumably the underlying OCI APIs) you can create an empty db home, then create a database into it as two distinct steps, while the Terraform OCI provider requires the db home resource to include a database.

I would have expected to be able to create an `oci_database_db_home` with no `database{}` object, then create an `oci_database_database` linked to the dbhome through the `db_home_id` argument.  But this is not possible since the `database{}` object is mandatory.

Any particular (design) reason why this is the case, or is this a bug?

**Comment by govindrao55** [comment]
Hi @lucianf , this is a good question. This is by design currently and you may consider this as a limitation in the provider. 
Partly due to how terraform works - it's stateful in nature and there is no direct one-to-one mapping between the `API`'s and terraform `resources`/`data sources` as it is between `SDK`/`CLI`/`Console` and `API`s. In the process of getting the same resource representation to handle all `create` and `update` scenarios, it gets challenging for a highly polymorphic resource such as `oci_database_db_home` and some deviations occur. I will have to dig deep and also check with rest of the team to find the reasons. I will respond here once I have the information.

**Comment by lucianf** [comment]
As a follow up, I'd kindly suggest you to consider fixing this, as this is a current shortcoming of the Terraform resource model.  There are use cases that are blocked by not being able to work with db homes and databases separately - for example, provisioning a (target) empty db home is required for the standby database of a Data Guard association.  Currently the only way to get this empty db home is via other approaches (Console/CLI/SDK/API).

Working with db homes may be a specific case that only applies to ExaCS (and presumably also BM DBCS), but it should be supported.
++ @alexng-canuck

**Comment by ravinitp** [comment]
We are very sorry that we couldn't respond to each and every issue reported on GitHub. 
Although we have refined the process to prioritize customer issues on GitHub, since this issue was reported a while ago, there is a good chance it may have been fixed in the latest version of Terraform Provider OCI. 
If you are still experiencing this issue, please create a new issue and label it as Bug.
